### ISSUE TITLE
Binance streaming stopped to work 

### ISSUE THREAD

**Issue by mahieddine** [bitrich-info/xchange-stream#140]
It seems like the binance streaming doesn't work anymore this is what i'm getting while running "BinanceManualExample.java" in the tests folder, the client just keep getting disconnected as soon as it gets connected
```
00:55:39.544 [main] DEBUG org.knowm.xchange.ExchangeFactory - Creating default exchange from class name
00:55:39.547 [main] DEBUG org.knowm.xchange.ExchangeFactory - Creating default exchange from class name
00:55:39.915 [main] DEBUG info.bitrich.xchangestream.binance.BinanceStreamingExchange - ExchangeMetaData [currencyPairs={}, currencies={}, publicRateLimits=null, privateRateLimits=null, shareRateLimits=false]
00:55:39.934 [main] DEBUG si.mazi.rescu.Config - Configuration from rescu.properties:
00:55:39.936 [main] DEBUG si.mazi.rescu.Config - httpConnTimeout = 30000
00:55:39.936 [main] DEBUG si.mazi.rescu.Config - httpReadTimeout = 30000
00:55:39.936 [main] DEBUG si.mazi.rescu.Config - proxyHost = null
00:55:39.936 [main] DEBUG si.mazi.rescu.Config - proxyPort = null
00:55:39.936 [main] DEBUG si.mazi.rescu.Config - ignoreHttpErrorCodes = false
00:55:39.966 [main] INFO  info.bitrich.xchangestream.binance.BinanceStreamingExchange - Calling Remote Init...
00:55:39.973 [main] DEBUG si.mazi.rescu.HttpTemplate - Executing GET request at https://api.binance.com/api/v1/exchangeInfo
00:55:41.490 [main] DEBUG si.mazi.rescu.HttpTemplate - Request http status = 200
00:55:42.441 [main] DEBUG si.mazi.rescu.HttpTemplate - Executing GET request at https://api.binance.com/api/v1/ticker/allPrices
00:55:42.711 [main] DEBUG si.mazi.rescu.HttpTemplate - Request http status = 200
00:55:43.011 [main] INFO  info.bitrich.xchangestream.binance.BinanceStreamingService - Connecting to wss://stream.binance.com:9443/stream
00:55:44.484 [nioEventLoopGroup-2-1] INFO  info.bitrich.xchangestream.service.netty.WebSocketClientHandler - WebSocket Client connected!
00:55:44.485 [nioEventLoopGroup-2-1] INFO  info.bitrich.xchangestream.binance.BinanceStreamingService - Subscribing to channel ethbtc@ticker
00:55:44.538 [nioEventLoopGroup-2-1] INFO  info.bitrich.xchangestream.binance.BinanceStreamingService - Subscribing to channel ltcbtc@ticker
00:55:44.539 [nioEventLoopGroup-2-1] INFO  info.bitrich.xchangestream.binance.BinanceStreamingService - Subscribing to channel ltcbtc@depth
00:55:44.543 [nioEventLoopGroup-2-1] INFO  info.bitrich.xchangestream.service.netty.WebSocketClientHandler - WebSocket Client received closing
00:55:44.547 [nioEventLoopGroup-2-1] INFO  info.bitrich.xchangestream.service.netty.WebSocketClientHandler - WebSocket Client disconnected!
00:55:44.547 [nioEventLoopGroup-2-1] INFO  info.bitrich.xchangestream.binance.BinanceStreamingService - Reopening websocket because it was closed by the host
00:55:44.594 [nioEventLoopGroup-2-1] INFO  info.bitrich.xchangestream.binance.BinanceStreamingService - Connecting to wss://stream.binance.com:9443/stream
00:55:45.784 [nioEventLoopGroup-2-2] INFO  info.bitrich.xchangestream.service.netty.WebSocketClientHandler - WebSocket Client connected!
00:55:45.785 [nioEventLoopGroup-2-2] INFO  info.bitrich.xchangestream.binance.BinanceStreamingService - Resubscribing channels
00:55:45.785 [nioEventLoopGroup-2-2] INFO  info.bitrich.xchangestream.service.netty.WebSocketClientHandler - WebSocket Client received closing
00:55:45.786 [nioEventLoopGroup-2-2] INFO  info.bitrich.xchangestream.service.netty.WebSocketClientHandler - WebSocket Client disconnected!
```
How can i fix this ? thank you

**Comment by trojanflash** [comment]
This happens because of slash at the beginning. Quickfix attached, but i get the problem with encoding.
[binance-fix.zip](https://github.com/bitrich-info/xchange-stream/files/1823841/binance-fix.zip)

**Comment by daniel-cohen** [comment]
Noticed it yesterday too and couldn't figure out what I was doing wrong. The node.js library was connecting just find. Thanks for reporting.

**Comment by mahieddine** [comment]
thank you @trojanflash your patch fixed the connection issue but as you mentioned there is still this problem with the encoding (as it seems), maybe Binance changed the compression algorithm?

**Comment by trojanflash** [comment]
@mahieddine also fix the problem, see attach 
[fixed-problem-with-websockets.zip](https://github.com/bitrich-info/xchange-stream/files/1824714/fixed-problem-with-websockets.zip)

**Comment by trojanflash** [comment]
@daniel-cohen you can checkout changes from my forked repository (https://github.com/trojanflash/xchange-stream) and run BinanceManualExample.
There is some more changes there such as moving to xchange 4.3.4 ect... (code style a little bit corrupted)

**Comment by trojanflash** [comment]
@daniel-cohen 
> PS: With regards to your 1st fix, Isn't it just enough to strip the first "/":
` subs = subs .startsWith("/") ? subs .substring(1) : subs ;`

i think it is a little bit ugly

**Comment by mahieddine** [comment]
thank you @trojanflash it worked perfectly so if i understand correctly the patch for the netty part, we were reading the message before uncompressing it right?
Anyway once again thank you very much for your help, will you make a PR for that ?

**Comment by trojanflash** [comment]
@mahieddine welcome. There are a few changes in commit besides this fix, so i don't know whether my PR will be accepted.

**Comment by daniel-cohen** [comment]
@trojanflash Thanks for the fix! I had some time to check your fork today, got it working and extracted the code into a PR for this issue. Hope you don't mind.

**Comment by trojanflash** [comment]
@daniel-cohen no problem, glad to help

**Comment by dozd** [comment]
Thanks @daniel-cohen and @trojanflash

**Comment by xttam** [comment]
On what version was this merged? Was the fix released already? I'm still getting this error on 4.3.3-SNAPSHOT.

```
91743 [nioEventLoopGroup-2-2] INFO info.bitrich.xchangestream.service.netty.WebSocketClientHandler - WebSocket Client connected!
91744 [nioEventLoopGroup-2-2] INFO info.bitrich.xchangestream.binance.BinanceStreamingService - Resubscribing channels
91744 [nioEventLoopGroup-2-2] INFO info.bitrich.xchangestream.service.netty.WebSocketClientHandler - WebSocket Client received closing
91745 [nioEventLoopGroup-2-2] INFO info.bitrich.xchangestream.service.netty.WebSocketClientHandler - WebSocket Client disconnected!
91745 [nioEventLoopGroup-2-2] INFO info.bitrich.xchangestream.binance.BinanceStreamingService - Reopening websocket because it was closed by the host
91745 [nioEventLoopGroup-2-2] INFO info.bitrich.xchangestream.binance.BinanceStreamingService - Connecting to wss://stream.binance.com:9443/stream
```

**Comment by sekharsa82** [comment]
I also got the same error, frequency of getting this error is very high (atleast once in every 15/20 minutes). This is resulting inconsistent orderbook. Does anyone know whether fix was already applied. 

08:21:20.883 [nioEventLoopGroup-2-2] INFO  info.bitrich.xchangestream.service.netty.WebSocketClientHandler - WebSocket Client received closing
08:21:20.887 [nioEventLoopGroup-2-2] INFO  info.bitrich.xchangestream.service.netty.WebSocketClientHandler - WebSocket Client disconnected!
08:21:20.887 [nioEventLoopGroup-2-2] INFO  info.bitrich.xchangestream.binance.BinanceStreamingService - Reopening websocket because it was closed by the host
08:21:20.888 [nioEventLoopGroup-2-2] INFO  info.bitrich.xchangestream.binance.BinanceStreamingService - Connecting to wss://stream.binance.com:9443/stream
08:21:21.715 [nioEventLoopGroup-2-3] INFO  info.bitrich.xchangestream.service.netty.WebSocketClientHandler - WebSocket Client connected!
08:21:21.715 [nioEventLoopGroup-2-3] INFO  info.bitrich.xchangestream.binance.BinanceStreamingService - Resubscribing channels

Thank you, I appreciate any help regarding this error.
